### PR TITLE
chore: update share image

### PR DESCRIPTION
**Pull request by georgiee**
Related to #71 

Replace the share image. New one is this one.
sy-share-image

### CODE REVIEW

**Review by feedm3 — APPROVED** [review]

**Comment by georgiee** [comment]
Sharing does not work. The `og:image` url needs to be absolute while it's now relative. 
https://socialsharepreview.com/?url=https://satellytescom-gkmetaimage.gtsb.io/

Edit: can't be that hard. Right now siteUrl is static,
I can do this `${site.siteMetadata.siteUrl}${DEFAULT_SHARE_IMAGE}`; but it will never work on the preview stage.

I would like to set the siteUrl to the deployed url. There must be a `env` to get that url during build time to apply it instead of the default value:

```
siteMetadata: {
    siteUrl: 'https://satellytes.com',
},
```

The docs are poor, maybe @feedm3 you have done this with gatsby cloud before?

**Comment by feedm3** [comment]
There is no environment variable in Gatsby Cloud that we can use out of the box: https://support.gatsbyjs.com/hc/en-us/articles/360052322954-Environment-Variables-Specific-to-Gatsby-Cloud

But we can "predict" the preview url ourselves: The patter is `${DOMAIN}-${BRANCH_NAME}.gtsb.io`, where the `DOMAIN` and `BRANCH_NAME` only contains lower case chars (no numbers or special characters). Like for this branch, the domain is `https://satellytescom-gkmetaimage.gtsb.io/`.

If the `BRNACH_NAME` is master, we would use our predefined production url.

**Comment by georgiee** [comment]
good idea, let's try to do that within this PR before merging.

**Comment by feedm3** [comment]
Meta image is now working for all Gatsby Preview Deployments, localhost and prod.

File: gatsby-config.js [hunk 1]
```diff
@@ -1,10 +1,38 @@
...
+
+  const formattedBranchName = BRANCH_NAME.toLowerCase().replace(
+    /[^a-zA-Z]/gi,
+    '',
+  );
+
+  return `https://${DOMAIN_NAME}-${formattedBranchName}.gtsb.io/`;
+};
+
+const BASE_URL = extractGatsbyCloudPreviewUrl() || 'http://localhost:8000';
```
Comment (georgiee) [inline gatsby-config.js:27]: 😍 Thank you

File: src/components/seo.tsx [hunk 2]
```diff
@@ -38,7 +38,8 @@ const SEO: React.FC<SeoProps> = ({
 
   const metaDescription = description || site.siteMetadata.description;
   const typeOfSite = siteType || 'website';
-  const imgUrl = imageUrl || DEFAULT_IMAGE_URL;
+  const metaImageUrl =
+    imageUrl || site.siteMetadata.siteUrl + DEFAULT_META_IMAGE_URL_PATH;
```
Comment (georgiee) [inline src/components/seo.tsx:42]: 👌

**Review by georgiee — COMMENTED** [review]
Thanks for the update. I'm going to merge this feature now

**Pull request merged**